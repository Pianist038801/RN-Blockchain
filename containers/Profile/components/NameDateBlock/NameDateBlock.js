import React from 'react'
import PropTypes from 'prop-types'
import { Text, View } from 'react-native'
import styles from '../../../../style.js'

const NameDateBlock = (props) => {
  const {name, line2, dateFrom, dateTo} = props

  return (
    <View style={styles.profileEntityContent}>
      <Text style={styles.profileEntityPrimaryText}>{name}</Text>
      <Text>{line2}</Text>
      <Text>{dateFrom} - {dateTo}</Text>
    </View>
  )
}

NameDateBlock.propTypes = {
  name: PropTypes.string.isRequired,
  line2: PropTypes.string.isRequired,
  dateFrom: PropTypes.string.isRequired,
  dateTo: PropTypes.string.isRequired
}

export default NameDateBlock
