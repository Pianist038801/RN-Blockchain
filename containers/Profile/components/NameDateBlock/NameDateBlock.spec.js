import React from 'react'
import renderer from 'react-test-renderer'
import NameDateBlock from './index'

describe('containers/Profile/components/NameDateBlock', () => {
  it('renders correctly', () => {
    const props = {
      name: 'hello',
      line2: 'Lorem ipsum',
      dateFrom: '2017-11-04',
      dateTo: '2018-01-24'
    }

    const tree = renderer.create(
      <NameDateBlock {...props} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
