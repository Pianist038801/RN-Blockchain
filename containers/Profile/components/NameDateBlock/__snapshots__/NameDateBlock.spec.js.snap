// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Profile/components/NameDateBlock renders correctly 1`] = `
<View
  style={
    Object {
      "marginLeft": 22.5,
      "width": 360,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#569cee",
        "fontWeight": "bold",
      }
    }
  >
    hello
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Lorem ipsum
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    2017-11-04
     - 
    2018-01-24
  </Text>
</View>
`;
